### ISSUE TITLE
DataTable: unselecting a row, seems not to work even in the docs examples

### ISSUE THREAD

**Issue by reksc** [primefaces/primevue#3433]
### Describe the bug

As we can read in the documentation under https://primefaces.org/primevue/datatable/selection:

> In single mode, a row is selected on click event of a row. **If the row is already selected then the row gets unselected**.

I tried multiple times, but on the docs page I can't unselect a row by clicking on it. The only scenario where it works is **_Multiple Selection without MetaKey_**.

It's possible, that I may have misunderstood the intentions, but _If the row is already selected then the row gets unselected_ seems pretty straightforward :).

### Reproducer

https://codesandbox.io/s/primevue-3-issue-template-ip3vm

### PrimeVue version

3.21.0

### Vue version

3.x

### Language

ALL

### Build / Runtime

Vue CLI App

### Steps to reproduce the behavior

Navigate to https://primefaces.org/primevue/datatable/selection and click on any table row to select it. Now click again. It's still selected, nothing happens.

### Expected behavior

A selected row should be unselected after clicking on it.

**Comment by tugcekucukoglu** [comment]
You also need to check `metaKeySelection` property https://primefaces.org/primevue/datatable. Also, we will check the documentation in Q1 of 2023.

Thanks for your attention.

**Comment by reksc** [comment]
That fixed the issue.

**tl;dr** In single mode, if I set `metaKeySelection` to `false`, unselecting a row starts working.

Many thanks for such a great library, I really enjoy working with it.
